### DISCUSSION TITLE
Don't show ajax button in list operation

### DISCUSSION THREAD

**Discussion by amigolj** [Laravel-Backpack/community-forum#1163] [Q&A (Help)] [answered]
Hi, help me please. I try add ajax button follow the documentation [here](https://backpackforlaravel.com/docs/6.x/crud-buttons#create-a-quick-button-with-ajax), but it doesn't show.
What I'm doing wrong?

Here my code in CRUD

` protected function setupListOperation()
    {

        CRUD::enableDetailsRow();
        CRUD::setDetailsRowView('feedback-details');
        
        CRUD::button('type')->stack('line')->view('crud::buttons.quick')->meta([
                'label' => 'Email',
                'icon' => 'la la-envelope',
                'wrapper' => [
                    'href' => function ($entry, $crud) {
                        return backpack_url("invoice/email");
                    },
                    'ajax' => true, // <- just add `ajax` and it's ready to make ajax request
                ]]
        );
        CRUD::column('description')->type('text');
        CRUD::column('status')->type('text');
        /**
         * Columns can be defined using the fluent syntax:
         * - CRUD::column('price')->type('number');
         */
    }`

php artisan backpack:version:
`### PHP VERSION:
8.3.11

### PHP EXTENSIONS:
Core, date, libxml, openssl, pcre, sqlite3, zlib, bcmath, bz2, calendar, ctype, curl, dom, hash, fileinfo, filter, ftp, gd, gettext, gmp, json, iconv, SPL, intl, session, mbstring, standard, mysqlnd, pcntl, exif, mysqli, PDO, PDO_Firebird, pdo_mysql, pdo_pgsql, pdo_sqlite, pgsql, Phar, posix, random, readline, Reflection, imap, shmop, SimpleXML, snmp, soap, sockets, sysvmsg, sysvsem, sysvshm, tidy, tokenizer, xml, xmlreader, xmlwriter, xsl, zip, igbinary, trader, redis, pdo_sqlsrv, dbase, protobuf, msgpack, pdo_dblib, phalcon, xhprof, mongodb, grpc, mailparse, sqlsrv, yaml, imagick, timezonedb, ldap, ssh2, xlswriter, memcached, sodium, memcache, mcrypt, ds

### LARAVEL VERSION:
10.48.17.0

### BACKPACK PACKAGE VERSIONS:
backpack/basset: 1.3.5
backpack/crud: 6.7.21
backpack/generators: v4.0.5
backpack/pro: 2.2.10
backpack/theme-tabler: 1.2.11
`

**Comment by karandatwani92** [comment, marked as answer]
Hey @amigolj 

Sorry for the inconvenience. **Docs** need a minor fix there. It is missing an `access` attribute, which tells if the user has permission to access it or not.

```diff
CRUD::button('type')->stack('line')->view('crud::buttons.quick')->meta(
    [
+        'access' => true, // or 'access' => 'operation_name', ex: list, update
        'label' => 'Email',
        'icon' => 'la la-envelope',
        'wrapper' => [
            'href' => function ($entry, $crud) {
                return backpack_url("invoice/email");
            },
            'ajax' => true, // <- just add `ajax` and it's ready to make ajax request
        ]
    ]
);
```

**Reply by amigolj** [reply]
Thank you, it works:)

**Comment by amigolj** [comment]
@karandatwani92  I don't know but now after press button, I got error 404. I need to add maybe some route?
![image](https://github.com/user-attachments/assets/f21622e7-ea84-429e-a29b-91276615f17f)
I added only this code for ajax button
![image](https://github.com/user-attachments/assets/3c1dd7e9-2546-4f94-8156-eb55557d753b)

**Reply by karandatwani92** [reply]
Try this tutorial:
https://backpackforlaravel.com/articles/tutorials/how-to-create-an-ajax-operation-with-quick-button

**Reply by amigolj** [reply]
> Try this tutorial: https://backpackforlaravel.com/articles/tutorials/how-to-create-an-ajax-operation-with-quick-button

Thanks it works:)
